core/embed/projects/prodtest/cmd/common.c

```diff
@@ -0,0 +1,416 @@
+#include "common.h"
+
+#include "buffer.h"
+#include "der.h"
+#include "ecdsa.h"
+#include "ed25519-donna/ed25519.h"
+#include "memzero.h"
+#include "nist256p1.h"
+#include "sha2.h"
+#include "string.h"
+
+// Identifier of context-specific constructed tag 3, which is used for
+// extensions in X.509.
+#define DER_X509_EXTENSIONS 0xa3
+
+// Identifier of context-specific primitive tag 0, which is used for
+// keyIdentifier in authorityKeyIdentifier.
+#define DER_X509_KEY_IDENTIFIER 0x80
+
+// DER-encoded object identifier of the authority key identifier extension
+// (id-ce-authorityKeyIdentifier).
+const uint8_t OID_AUTHORITY_KEY_IDENTIFIER[] = {0x06, 0x03, 0x55, 0x1d, 0x23};
+
+// clang-format off
+static const uint8_t ECDSA_P256_WITH_SHA256[] = {
+  0x30, 0x13, // a sequence of 19 bytes
+    0x06, 0x07, // an OID of 7 bytes
+      0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02, 0x01, // corresponds to ecPublicKey in X.509
+    0x06, 0x08, // an OID of 8 bytes
+      0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07, // corresponds to prime256v1 in X.509
+};
+
+static const uint8_t EDDSA_25519[] = {
+  0x30, 0x05, // a sequence of 5 bytes
+    0x06, 0x03, // an OID of 3 bytes
+        0x2b, 0x65, 0x70, // corresponds to EdDSA 25519 in X.509
```

Comment:
nit: less indentation

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/5162/commits/e50c85332765ad738f31f64d0c02aee4990e16f6.